### ISSUE TITLE
分类任务id标签不连续产生的问题

### ISSUE THREAD

**Issue by lln1997** [modelscope/ms-swift#9539]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new question or discussion topic. / 我已经搜索过现有的 issues，确认这是一个新的问题与讨论。

### Question Description / 问题描述

# If `num_labels` is provided, it will be considered a classification task,
# and AutoModelForSequenceClassification will be used to load the model.
# You can also specify `--model Qwen/Qwen2.5-0.5B-Instruct --use_chat_template true`.
NPROC_PER_NODE=8 \
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
swift sft \
    --model /opt/tiger/bert_base_chinese/Qwen3.5-2B\
    --tuner_type full \
    --dataset '' \
    --load_from_cache_file true \
    --split_dataset_ratio 0 \
    --torch_dtype bfloat16 \
    --num_train_epochs 5 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 4 \
    --learning_rate 1e-4 \
    --lora_rank 8 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --gradient_accumulation_steps 1 \
    --save_strategy epoch \
    --save_total_limit 2 \
    --logging_steps 5 \
    --max_length 8192 \
    --output_dir  \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --num_labels 47 \
    --task_type seq_cls \
    --use_chat_template false

其中我的num_labels为47，最高id是到46，但是实际上中间有id缺失，比如没有19，训练报错：

<img width="1330" height="1006" alt="Image" src="https://github.com/user-attachments/assets/947078b1-cd3c-4a2e-9363-ba4c9fe9c58b" />
请问下标签一定要连续嘛，不连续的话怎么解决呀

**Comment by Jintao-Huang** [comment]
数据集格式 是怎么样的

是符合自定义数据集文档中的 分类任务吗

**Comment by lln1997** [comment]
> 数据集格式 是怎么样的
> 
> 是符合自定义数据集文档中的 分类任务吗

{"messages": [{"role": "user", "content": "aaaaa"}], "label": 0}
{"messages": [{"role": "user", "content": "aaaaa"}], "label": 1}
{"messages": [{"role": "user", "content": "aaaaa"}], "label": 8}
这样的，符合的，我把id填满就没这个报错了，比如num_labels=8，如果缺少5或者其中任意几个id就报错应该。但我觉得缺少值应该也不能报错

**Comment by Jintao-Huang** [comment]
"label": 8

那么 num_labels应该设置成9 啊

**Comment by lln1997** [comment]
> "label": 8
> 
> 那么 num_labels应该设置成9 啊

哦哦我是举个例子，实际我是设的比最大label+1。但看起来好像你们觉得不应该产生这个问题，可能才会质疑是我哪个地方设置错了嘛

**Comment by Jintao-Huang** [comment]
我这里测试是正常 没问题的。这里设置成 num_labels 9

> > 数据集格式 是怎么样的
> > 是符合自定义数据集文档中的 分类任务吗
> 
> {"messages": [{"role": "user", "content": "aaaaa"}], "label": 0} {"messages": [{"role": "user", "content": "aaaaa"}], "label": 1} {"messages": [{"role": "user", "content": "aaaaa"}], "label": 8} 这样的，符合的，我把id填满就没这个报错了，比如num_labels=8，如果缺少5或者其中任意几个id就报错应该。但我觉得缺少值应该也不能报错

**Comment by lln1997** [comment]
> 我这里测试是正常 没问题的。这里设置成 num_labels 9我这里测试是正常 没问题的。这里设置成 num_labels 9
> 
> > > 数据集格式 是怎么样的 数据集格式 是怎么样的
> > > 是符合自定义数据集文档中的 分类任务吗 是符合自定义数据集文档中的 分类任务吗
> > 
> > 
> > {"messages": [{"role": "user", "content": "aaaaa"}], "label": 0} {"messages": [{"role": "user", "content": "aaaaa"}], "label": 1} {"messages": [{"role": "user", "content": "aaaaa"}], "label": 8} 这样的，符合的，我把id填满就没这个报错了，比如num_labels=8，如果缺少5或者其中任意几个id就报错应该。但我觉得缺少值应该也不能报错{"messages": [{"role": "user", "content": "aaaaa"}], "label": 0} {"messages": [{"role": "user", "content": "aaaaa"}], "label": 1} {"messages": [{"role": "user", "content": "aaaaa"}], "label": 8} 这样的，符合的，我把id填满就没这个报错了，比如num_labels=8，如果缺少5或者其中任意几个id就报错应该。但我觉得缺少值应该也不能报错

明白，感谢，我再检查一下
